extensions/attention-stream/scripts/generate-manifest.ts

```diff
@@ -3,73 +3,82 @@ import fs from "fs/promises";
 import minimist from "minimist";
 import { sys } from "typescript";
 
-const args = (minimist(process.argv.slice(2)));
+const args = minimist(process.argv.slice(2));
 
 async function main(manifestVersion, browser) {
-    const manifest = manifestTemplate;
-    manifest["manifest_version"] = manifestVersion;
+  const manifest = manifestTemplate;
+  manifest["manifest_version"] = manifestVersion;
 
-    if (!(["firefox", "chrome"].includes(browser))) {
-        throw new Error(`unknown browser: ${browser}`);
-    }
+  if (!["firefox", "chrome"].includes(browser)) {
+    throw new Error(`unknown browser: ${browser}`);
+  }
+
+  if (browser === "firefox") {
+    manifest["browser_specific_settings"] = {
+      gecko: {
+        id: "attention-stream@rally.mozilla.org",
+        strict_min_version: "101.0",
+      },
+    };
+  }
+
+  switch (manifestVersion) {
+    case 2: {
+      manifest["browser_action"] = {
+        default_title: "Rally Attention Stream",
+      };
+
+      manifest["background"] = {
+        scripts: ["dist/background.js"],
+      };
+
+      manifest["web_accessible_resources"] = ["dist/youtube/yt.injected.js"];
+
+      manifest["permissions"].push("<all_urls>");
 
-    if (browser === "firefox") {
-        manifest["browser_specific_settings"] = {
-            gecko: {
-                id: 'attention-stream@rally.mozilla.org',
-                strict_min_version: '101.0'
-            }
-        }
+      break;
     }
 
-    switch (manifestVersion) {
-        case 2: {
-            manifest["browser_action"] = {
-                "default_title": "Rally Attention Stream"
-            }
-
-            manifest["background"] = {
-                "scripts": [
-                    "dist/background.js"
-                ]
-            }
-            manifest.permissions.push("<all_urls>");
-
-            break;
-        }
-        case 3: {
-            manifest["action"] = {
-                "default_title": "Rally Attention Stream"
-            }
-
-            manifest["background"] = {
-                "service_worker": "dist/background-loader.js"
-            }
-
-            manifest["host_permissions"] = ["<all_urls>"];
-
-            break;
-        }
-        default: {
-            throw new Error("unknown manifest version");
-        }
+    case 3: {
+      manifest["action"] = {
+        default_title: "Rally Attention Stream",
+      };
+
+      manifest["background"] = {
+        service_worker: "dist/background-loader.js",
+      };
+
+      manifest["web_accessible_resources"] = [
+        {
+          resources: ["dist/youtube/yt.injected.js"],
```

Comment:
You might be able to avoid having to expose this to content by using the [new-ish injection API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/scripting/executeScript).

Reply:
ah I forgot to update you on this. After your Slack message I actually went and tried to do this. But after implementing it, I realized that scripts injected via `executeScript` live in the isolated content script environment. And the content script environment has access to the DOM, but it **doesn't have access** to the same global scope of variables that the page does. So the monkeypatching doesn't work: replacing `window.fetch` has no effect on the page's `window.fetch`. So this script needs to be injected in this round-about way, so that the script is actually put directly into the page, rather than living in an isolated environment.

Reply:
filed an issue for this: https://github.com/mozilla-rally/rally/issues/228